pkg/operator/ceph/controller/predicate.go

```diff
@@ -284,3 +355,98 @@ func isUpgrade(oldLabels, newLabels map[string]string) bool {
 
 	return false
 }
+
+func isCanary(obj runtime.Object) bool {
+	// If not a deployment, let's not reconcile
+	d, ok := obj.(*appsv1.Deployment)
+	if !ok {
+		return false
+	}
+
+	// Get the labels
+	labels := d.GetLabels()
+
+	labelVal, labelKeyExist := labels["mon_canary"]
+	if labelKeyExist && labelVal == "true" {
+		logger.Debugf("do not reconcile %q on monitor canary deployments", d.Name)
+		return true
+	}
+
+	return false
+}
+
+// IsHotPlugCM informs whether the object is the cm for hot-plug disk
+func IsHotPlugCM(obj runtime.Object) bool {
+	// If not a ConfigMap, let's not reconcile
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return false
+	}
+
+	// Get the labels
+	labels := cm.GetLabels()
+
+	labelVal, labelKeyExist := labels[k8sutil.AppAttr]
+	if labelKeyExist && labelVal == discoverDaemon.AppName {
+		return true
+	}
+
+	return false
+}
+
+func isCMToIgnoreOnUpdate(obj runtime.Object) bool {
```

Comment:
Which CMs do we actually want to trigger the reconcile? Seems like the only one worth watching might be rook-ceph-csi-config. All the others either have another watcher or we can't recover them if they get changed. Should we just check for an inclusion list instead of exclusion list?

Reply:
`rook-ceph-csi-config` is not even managed by the watcher because its owerref is on the Operator deployment, not the CephCluster CR but yes let me look into an inclusion list and see if we can do better.